pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go

```diff
@@ -81,12 +82,24 @@ func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataS
 		return nil, err
 	}
 
-	return &tsdb.Response{
-		Results: map[string]*tsdb.QueryResult{
-			"": {
-				Dataframes: pbRes.Frames,
-				Meta:       simplejson.NewFromAny(pbRes.Metadata),
-			},
-		},
-	}, nil
+	tR := &tsdb.Response{
```

Comment:
Nit, but no biggie. This code until line 105 is duplicated and could be extracted to a shared function.

Reply:
Assuming you mean the duplication in Transform, in which case I think maybe simpler to keep that separate for now (applying rule of 3, I guess wrongly, since they are identical, but seems like the right thing here to me right now).